Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/DownloadDiagnosisKeysTask.kt

```diff
@@ -198,13 +198,13 @@ class DownloadDiagnosisKeysTask @Inject constructor(
             DateTimeZone.UTC
         )
         return (LocalData.lastTimeDiagnosisKeysFromServerFetch() == null ||
-            currentDate.withTimeAtStartOfDay() != lastFetch.withTimeAtStartOfDay()).also {
+            currentDate.isAfter(lastFetch)).also {
```

Comment:
What effect will this have? Won't this always be try, because unless we have a sub-milliseconds racecondition, `timeStamper.nowUTC` will always be after `LocalData.lastTimeDiagnosisKeysFromServerFetch()`.

Reply:
i agree. what harm would it do if we just delete the complete condition?

Reply:
I think it's already effectively removed, because `currentDate.isAfter(lastFetch)`  will almost always return true.

Reply:
Yes, my thinking too. The check solely protects us from someone time traveling backward.
IMHO all these checks should not be implemented by the Task but rather by the KeyPackageSyncTool.

Reply:
Yes, I tested again and it seems it is always returning true.

Reply:
Removed it entirely while ensuring that the task is only being run once every hour.